### DISCUSSION TITLE
Accelerate OpenRadioss with OpenACC compiler directives?

### DISCUSSION THREAD

**Discussion by ZhengZQ007** [OpenRadioss/OpenRadioss#1713] [Q&A]
The NVIDIA HPC SDK Fortran compiler support GPU acceleration with OpenACC compiler directives, does anyone try to build a GPU-accelerated OpenRadioss?

**Comment by elequiniou** [comment]
Dear Zheng,

This would be cool! 
Indeed, this is still a challenge for legacy explicit code like Radioss to efficiently use GPU.  With progress made on tools, like OpenACC and OpenMP, on hardware with faster communication between CPU and GPU and coherent memory, and this possibility to collaborate efficiently altogether with open-source, this is a challenge that could be addressed and hopefully solved in the future!
If you or others have some ideas, I'd be happy to exchange further.

Thank you,

Eric

**Reply by ZhengZQ007** [reply]
Hi, Eric. In OpenRadioss source code, there are many OpenMP compiler directives aim to efficiently use CPU. Due to my little knowledge to OpenACC, I have a simple idea that can we just modify these OpenMP compiler directives by appropriate OpenACC compiler directives. I have not verify will it be applicable.

**Comment by elequiniou** [comment]
Unfortunately, this won't work. The way it is parallelized needs to be completely reviewed, not an OpenACC issue, it would be the same with OpenMP 5.0 offload. The approach to offload on GPU is different from loop parallelization with classic OpenMP programing.
Additionally, memory transfer between CPU & GPU needs to be minimized.

**Reply by ZhengZQ007** [reply]
Thanks very much for your answer.
